cmd/nerdctl/run.go

```diff
@@ -966,6 +966,35 @@ func parseKVStringsMapFromLogOpt(cmd *cobra.Command, logDriver string) (map[stri
 		if _, ok := logOptMap[logging.MaxSize]; !ok {
 			delete(logOptMap, logging.MaxFile)
 		}
+		for _, value := range logging.FluentdLogOpts {
+			if _, ok := logOptMap[value]; ok {
+				logrus.Warnf("log-opt %s is ignored for json-file log driver", value)
+			}
+		}
+		for _, value := range logging.JournalDriverLogOpts {
+			if _, ok := logOptMap[value]; ok {
+				logrus.Warnf("log-opt %s is ignored for json-file log driver", value)
+			}
+		}
+	}
+	if logDriver == "journald" {
+		for _, value := range logging.JSONDriverLogOpts {
+			if _, ok := logOptMap[value]; ok {
+				logrus.Warnf("log-opt %s is ignored for journald log driver", value)
+			}
+		}
+		for _, value := range logging.FluentdLogOpts {
+			if _, ok := logOptMap[value]; ok {
+				logrus.Warnf("log-opt %s is ignored for journald log driver", value)
+			}
+		}
+	}
```

Comment:
```suggestion
	if logDriver == "journald" {
		for opt, _ := range logOptMap {
			if _, ok := logging.JournalDriverLogOpts[opt]; !ok {
				logrus.Warnf("log-opt %s is ignored for journald log driver", opt)
			}
		}
	}
```

this need to convert this (https://github.com/containerd/nerdctl/pull/1244/files#diff-eb19031b20f577302a6bfbcbb42ee60f37256f9ed2044d905a7aa681309bca69R40)  to a map 

```
var FluentdLogOpts  = map[string]struct{}{}
```

Reply:
Oh, I got your point, but I think this covert would be unnecessary? I guess?

`xxxLogOpts` could just be a slice to store the option key.

But your way seems more friendly to read, I'm not sure about this

Reply:
An other ++ of my suggestion is that we avoid changing this Block each time we add new driver

Reply:
I agree, `map[string]struct{}{}` or `[]string` are both ok, few parameters will not cause any performance issues